tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/DirectoryHelper.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.IO;
+
+namespace Azure.Sdk.Tools.TestProxy.Common
+{
+    public static class DirectoryHelper
+    {
+        /// <summary>
+        /// Recursively delete a git directory. Calling Directory.Delete(path, true), to recursiverly delete a directory
+        /// that was populated from sparse-checkout, will fail. This is because the git files under .git\objects\pack 
+        /// have file attributes on them that will cause an UnauthorizedAccessException when trying to delete them. In order
+        /// to delete it, the file attributes need to be set to Normal.
+        /// </summary>
+        /// <param name="directory">The git directory to delete</param>
+        public static void DeleteGitDirectory(string directory)
+        {
+            File.SetAttributes(directory, FileAttributes.Normal);
+
+            string[] files = Directory.GetFiles(directory);
+            string[] dirs = Directory.GetDirectories(directory);
+
+            foreach (string file in files)
+            {
+                File.SetAttributes(file, FileAttributes.Normal);
```

Comment:
How slow is this if we aim it at a big cloned repo? Does the individual setting of attributes visibly slow this? 

If so, we could have a pre-check that ensures the files is within a `.git` directory before applying this logic, otherwise normal delete?

Reply:
I ran the same C# code deleting an entire Azure/azure-sdk-for-java enlistment, it took 37 seconds. I also ran a powershell script that just simply called `Remove-Item -LiteralPath <ThePath> -Force -Recurse` on an entire Azure/azure-sdk-for-java enlistment which took 26 seconds. Being that the assets repository should only have a fraction of the number of files that the language repository has (since it's only assets), I'm not worried about this taking undue amounts of time.